File: source/strategy/js/ducks/Ducks/Duck.js

var Duck = function() {
	this.display = function() { 

	};

	this.swim = function() { 
		console.log("I am swim like duck!"); 
	};

	this.fly = function() { 
		this.flyBehavior.fly(); 
	};

	this.sound = function() { 
		this.soundBehavior.sound(); 
	};

	this.setFlyBehavior = function(behavior) {
		this.flyBehavior = behavior;
	}
};

File: source/strategy/java/Duck.java

public abstract class Duck {
	public abstract void display();

	public SoundBehavior soundBehavior;
	public void setSoundBehavior(SoundBehavior soundBehavior) {
		this.soundBehavior = soundBehavior;
	}

	public void swim() {
		System.out.println("I am swim like decoy!");
	}

	public void sound() {
		this.soundBehavior.sound();
	}
}